nixos/modules/services/continuous-integration/github-runner/service.nix

```
@@ -153,18 +157,21 @@ with lib;
             ln -s '${newConfigPath}' "${currentConfigPath}"
           fi
         '';
-        setupRuntimeDir = writeScript "setup-runtime-dirs" ''
+        setupWorkDir = writeScript "setup-work-dirs" ''
+          # Cleanup previous service
+          ${pkgs.findutils}/bin/find "$WORK_DIRECTORY" -mindepth 1 -delete
```

We need to pass the `-H` option for the case that `$WORK_DIRECTORY` is a symlink. Otherwise, the directory won't be cleaned causing the script to fail in the following steps:

```suggestion
          ${pkgs.findutils}/bin/find -H "$WORK_DIRECTORY" -mindepth 1 -delete
```

Note that the flag will instruct `find` only to follow symlinks for the given `"$WORK_DIRECTORY"` argument, not for any symlinks within that directory. For details, see _find(1)_.

---

Good catch, thanks!